src/sinks/util/request_builder.rs

```diff
@@ -1,10 +1,24 @@
-use std::io;
+use std::{io, num::NonZeroUsize};
 
 use bytes::Bytes;
 use vector_common::request_metadata::{GroupedCountByteSize, RequestMetadata};
 
 use super::{encoding::Encoder, metadata::RequestMetadataBuilder, Compression, Compressor};
 
+pub fn default_request_builder_concurrency_limit() -> NonZeroUsize {
```

Comment:
It's not particularly important, but do you think it might be nicer to create a `ConcurrencyLimit` wrapper type and implement `Default` for that?